### PR TITLE
forklift_versions: allow specifying an offset to get an older version

### PR DESCRIPTION
**Pull request by evgeni**
this can be used to install a version n server with an n-1 proxy

### CODE REVIEW

**Comment by ehelms** [comment]
I think I may need to see this in action to understand how it will be used.

**Comment by evgeni** [comment]
@ehelms like this?

File: docs/testing.md [hunk 1]
```diff
@@ -78,37 +79,9 @@ When you are finished with the test, you can tear down the associated infrastruc
 #### Examples
 
     ansible-playbook pipelines/install_pipeline.yml -e forklift_state=up -e pipeline_os=debian10 -e pipeline_type=foreman -e pipeline_version=nightly
+    ansible-playbook pipelines/install_pipeline.yml -e forklift_state=up -e pipeline_os=centos7 -e pipeline_type=katello -e pipeline_version=nightly -e pipeline_proxy_delay=1
```
Comment (ehelms) [inline docs/testing.md:82]: What do you think about: 

 * pipeline_proxy_version_offset=1
 * pipeline_proxy_versions_back=1

Comment (evgeni) [reply docs/testing.md]: I like the version offset one, used it now.

**Comment by ehelms** [comment]
This probably complicates things, can we modify the box name on spin up if the offset is set? That way I have clearer understanding of the version of the proxy deployed? e.g.

```
changed: [pipe-katello-server-nightly-centos7]
changed: [pipe-katello-proxy-3.18-centos7]
```

**Comment by evgeni** [comment]
No, the box name generation happens before the role is executed.

The only way to achieve that if we *manually* pass the proxy version to the playbook like:

```console
$ ansible-playbook pipelines/install_pipeline.yml -e forklift_state=up -e pipeline_os=centos7 -e pipeline_type=katello -e pipeline_server_version=nightly -e pipeline_proxy_version=3.18
```

**Comment by ehelms** [comment]
What if we put the offset in the box name? `pipe-katello-proxy-nightly-1-centos7`  that kinda looks like nightly minus 1 :) or pipe-katello-proxy-offset-1-centos7

**Comment by evgeni** [comment]
and regular boxes would then be `nightly-0`?

**Comment by ehelms** [comment]
> and regular boxes would then be `nightly-0`?

I guess, I realize it is a big ugly. I am just worried when making use of this I will get myself confused debugging something about what is what version.

**Comment by evgeni** [comment]
Would a shiny `/etc/motd` help?

**Comment by ehelms** [comment]
> Would a shiny `/etc/motd` help?

Can it just say "ERIC THE VERSION OF THE PROXY IS X.Y - Evgeni" ?

**Comment by evgeni** [comment]
Yeah, something like that ;)

**Comment by evgeni** [comment]
The hostnames we generate are already quite long and I don't think adding a single digit would make the difference obvious - but adding a longer string will bring issues with too long names.

**Comment by ehelms** [comment]
Since there is not a rather good solution for this, let's give this a try as is and see how often we bump into said wall.

**Comment by evgeni** [comment]
going to close this and replace it with something else.

**Pull request closed without merging**